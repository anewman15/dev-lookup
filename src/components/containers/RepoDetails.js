import { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { fetchRepoDetails } from '../../sandbox/fetchData';
import Loading from '../presentational/Loading';
import SomethingWentWrong from '../presentational/SomethingWentWrong';

const RepoDetails = ({ match }) => {
  const [repoDetails, setRepoDetails] = useState({});
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState(false);
  const uri = `https://api.github.com/repos/${match.params.repo_owner}/${match.params.repo_name}`;

  const args = {
    setRepoDetails,
    setIsLoading,
    setError,
    uri,
  };

  useEffect(() => {
    fetchRepoDetails(args);
  }, []);

  return (
    <div>
      {isLoading ? <Loading /> : null}
      {error ? <SomethingWentWrong /> : null}
      <div>
        <p>{repoDetails.full_name}</p>
        <a href={repoDetails.html_url}>{repoDetails.html_url}</a>
        <p>{repoDetails.language}</p>
        <p>{repoDetails.description ? repoDetails.description : 'No description available'}</p>
        <p>
          Created: &nbsp;
          {repoDetails.created_at}
        </p>
        <p>
          Last updated: &nbsp;
          {repoDetails.updated_at}
        </p>
        <p>
          Stars: &nbsp;
          {repoDetails.stargazers_count}
        </p>
        <p>
          Watchers: &nbsp;
          {repoDetails.watchers_count}
        </p>
        <p>
          Subscribers: &nbsp;
          {repoDetails.subscribers_count}
        </p>
        <hr />
      </div>
    </div>
  );
};

RepoDetails.propTypes = {
  index: PropTypes.number,
  repo: PropTypes.object,
}.isReuired;

export default withRouter(RepoDetails);
