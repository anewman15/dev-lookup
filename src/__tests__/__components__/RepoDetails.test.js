import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import repos from '../../sandbox/repos';
import RepoDetails from '../../components/containers/RepoDetails';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('RepoDetails', () => {
  it('renders the repository div', () => {
    const tree = renderer
      .create(<RepoDetails repoDetails={JSON.parse(repos)[0]} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
