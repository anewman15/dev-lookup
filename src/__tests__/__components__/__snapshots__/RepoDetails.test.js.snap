// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoDetails renders the repository div 1`] = `
<div
  className="container"
>
  <div
    className="repoCard"
  >
    <div
      className="flexBetween"
    >
      <div
        className="flexBetween"
      >
        <p>
          Language: Ruby
        </p>
        <p
          className="repoSpecs"
        >
          Stars: 7
        </p>
      </div>
      <div
        className="flexBetween"
      >
        <p
          className="repoSpecs"
        >
          Watchers: 7
        </p>
        <p>
          Subscribers: undefined
        </p>
      </div>
    </div>
    <hr />
    <div>
      <div
        className="flexBetween"
      >
        <p>
          Repository Name: anewman15/activize
        </p>
        <p>
          Owner: anewman15
        </p>
      </div>
      <div>
        <a
          className="repoUrl"
          href="https://github.com/anewman15/activize"
          rel="noreferrer"
          target="_blank"
        >
          https://github.com/anewman15/activize
        </a>
      </div>
    </div>
    <hr />
    <div>
      <p>
        Description:
      </p>
      <p>
        This is an activity logger app that helps a user create daily activities and categorize them according to various groups they have in mind. They are able to name and create those groups according to their convenience.
      </p>
    </div>
    <hr />
    <div
      className="flexBetween"
    >
      <p>
        Created: 2020-11-03T10:54:13Z
      </p>
      <p>
        Last updated: 2021-02-26T15:07:02Z
      </p>
    </div>
    <hr />
  </div>
</div>
`;
